frame/bags-list/src/list/mod.rs

```diff
@@ -384,6 +384,77 @@ impl<T: Config> List<T> {
 		})
 	}
 
+	/// Put `heavier_id` to the position directly in front of `lighter_id`. Both ids must be in
+	/// same bag  and the `weight_of` `lighter_id` must be less than that of `heavier_id`
+	pub(crate) fn put_in_front_of(
+		lighter_id: &T::AccountId,
+		heavier_id: &T::AccountId,
+		weight_of: Box<dyn Fn(&T::AccountId) -> VoteWeight>,
+	) -> Result<(), crate::pallet::Error<T>> {
+		use crate::pallet;
+		use frame_support::ensure;
+
+		let lighter_node = Node::<T>::get(&lighter_id).ok_or(pallet::Error::IdNotFound)?;
+		let mut heavier_node = Node::<T>::get(&heavier_id).ok_or(pallet::Error::IdNotFound)?;
+
+		ensure!(lighter_node.bag_upper == heavier_node.bag_upper, pallet::Error::NotInSameBag);
+		ensure!(
+			heavier_node.next.as_ref() != Some(lighter_id),
+			pallet::Error::AlreadyHigherPosition
+		);
```

Comment:
I don't think this check brings any value. It is a bit ambiguous, since you are assuming an order: 

```
light_node -> heave_node
```

to signify that the order is already good. But for example: 

```
heavy_node -> x -> x -> light_node
```

to me is the same (meaning that there's nothing to be done). 

I would strongly recommend not having this error type at all. If you are already heavier than someone else, your execution is success, and you pay some fees for no reason for it.

Reply:
Also other suggestions: 

1. If you really think this error checking is worth keeping, let's call it `AlreadyInFront` to be more clear.
2. Why moving to the front and not the back? The back is a sensible one. Currently, the first member of each bag is basically invincible, which is not justified IMO.

Reply:
yeah I don't think its worth keeping, and its not really an intrinsic error for the blockchain, more just a user mistake like you say.

Reply:
60743ccfce9cb8a85d8a817e009d4a3bb21ff771